app/main.py

```diff
@@ -1,34 +1,139 @@
+from typing import List, Dict, Tuple
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int, int],
+            end: Tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        self.decks = []
+
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.is_drowned = is_drowned
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        dead_deck = self.get_deck(row, column)
+        dead_deck.is_alive = False
+        is_at_least_one_deck_alive: bool = False
+        for deck in self.decks:
+            if deck.is_alive:
+                is_at_least_one_deck_alive = True
+        if not is_at_least_one_deck_alive:
+            self.is_drowned = True
```

Comment:
no need to go through all the list of decks u can make loop and if some deck.is alive, is_at_least_one_deck_alive = True and break the loop